components/discussions/GithubDiscussions.tsx

```diff
@@ -0,0 +1,51 @@
+import GithubDiscussion from "@/components/discussions/GithubDiscussion";
+import { parseDateRangeSearchParam } from "@/lib/utils";
+import { ParsedDiscussion } from "@/scraper/src/github-scraper/types";
+
+interface Props {
+  discussions: ParsedDiscussion[];
+  searchParams?: { [key: string]: string };
+}
+
+const GithubDiscussions = ({ discussions, searchParams }: Props) => {
+  const category = searchParams?.category;
+  const dateRange = searchParams?.between;
+  const [start, end] = parseDateRangeSearchParam(dateRange);
```

Comment:
Not resolved yet...

Reply:
![image](https://github.com/user-attachments/assets/a4fad9d8-623f-4441-8ded-0c3d0444efea)

It is updated i keep the date range separate because it is required in if condition

Reply:
By using `parseDateRangeSearchParam` utility function, you will always have a date range. So that check can be avoided altogether...